pandas/sparse/tests/test_sparse.py

```diff
@@ -770,6 +770,32 @@ def test_combine_first(self):
         assert_sp_series_equal(result, result2)
         assert_sp_series_equal(result, expected)
 
+class TestSparseHandlingMultiIndexes(tm.TestCase):
+
+    def setUp(self):
+        miindex = pd.MultiIndex.from_product([["x","y"], ["10","20"]],names=['row-foo', 'row-bar'])
+        micol = pd.MultiIndex.from_product([['a','b','c'], ["1","2"]],names=['col-foo', 'col-bar'])
+        dense_multiindex_frame = pd.DataFrame(index=miindex, columns=micol).sortlevel().sortlevel(axis=1)
+        self.dense_multiindex_frame = dense_multiindex_frame.fillna(value=3.14)
+
+    def test_to_sparse_preserve_multiindex_names_on_columns(self):
+        sparse_multiindex_frame = self.dense_multiindex_frame.to_sparse()
+        self.assertTrue(self.dense_multiindex_frame.columns.equals(sparse_multiindex_frame.columns))
```

Comment:
this should all be 2 test and just use
assert_sp_frame_equal or dense version to compare versus and expected frame